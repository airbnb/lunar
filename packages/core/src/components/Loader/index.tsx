import React from 'react';
import withStyles, { WithStylesProps } from '../../composers/withStyles';

export type Props = {
  /** Display inline instead of absolutely positioned. */
  inline?: boolean;
  /** Invert colors. */
  inverted?: boolean;
  /** Increase the dot size. */
  large?: boolean;
  /** Position statically instead of absolutely. */
  static?: boolean;
};

/** A small 3-dot loading indicator. */
export class Loader extends React.Component<Props & WithStylesProps> {
  static defaultProps = {
    inline: false,
    inverted: false,
    large: false,
    static: false,
  };

  render() {
    const { cx, styles, inline, inverted, large, static: isStatic } = this.props;

    return (
      <div
        className={cx(
          styles.loader,
          inline && styles.loader_inline,
          !isStatic && !inline && styles.loader_absolute,
        )}
      >
        {[1, 2, 3].map(no => (
          <span
            key={no}
            className={cx(
              styles.dot,
              styles[`dot_${no}`],
              large && styles.dot_large,
              inverted && styles.dot_inverted,
            )}
          />
        ))}
      </div>
    );
  }
}

export default withStyles(({ color, unit }) => ({
  loader: {
    margin: '0 auto',
    textAlign: 'center',
  },

  loader_absolute: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translateX(-50%) translateY(-50%)',
  },

  loader_inline: {
    display: 'inline-block',
  },

  dot: {
    width: 6,
    height: 6,
    marginRight: unit / 2,
    borderRadius: '100%',
    display: 'inline-block',
    animationName: {
      name: 'fade',
      '0%, 80%, 100%': {
        opacity: 0,
      },
      '30%, 50%': {
        opacity: 1,
      },
    },
    animationDuration: '0.8s',
    animationIterationCount: 'infinite',
    animationTimingFunction: 'linear',
    animationFillMode: 'both',
    verticalAlign: 'middle',
    backgroundColor: color.core.primary[3],
  },

  dot_inverted: {
    backgroundColor: color.accent.bg,
  },

  dot_large: {
    width: 12,
    height: 12,
    marginRight: unit,
  },

  dot_1: {
    animationDelay: '-0.3s',
  },

  dot_2: {
    animationDelay: '-0.15s',
  },
}))(Loader);
