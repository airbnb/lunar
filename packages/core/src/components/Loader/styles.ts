import { StyleSheet } from '../../hooks/useStyles';

const styleSheet: StyleSheet = ({ color, unit }) => ({
  loader: {
    margin: '0 auto',
    textAlign: 'center',
  },

  loader_absolute: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translateX(-50%) translateY(-50%)',
  },

  loader_inline: {
    display: 'inline-block',
  },

  dot: {
    width: 6,
    height: 6,
    marginRight: unit / 2,
    borderRadius: '100%',
    display: 'inline-block',
    animationName: {
      name: 'fade',
      '0%, 80%, 100%': {
        opacity: 0,
      },
      '30%, 50%': {
        opacity: 1,
      },
    },
    animationDuration: '0.8s',
    animationIterationCount: 'infinite',
    animationTimingFunction: 'linear',
    animationFillMode: 'both',
    verticalAlign: 'middle',
    backgroundColor: color.core.primary[3],
  },

  dot_inverted: {
    backgroundColor: color.accent.bg,
  },

  dot_large: {
    width: 12,
    height: 12,
    marginRight: unit,
  },

  dot_1: {
    animationDelay: '-0.3s',
  },

  dot_2: {
    animationDelay: '-0.15s',
  },
});

export default styleSheet;

export { styleSheet };
