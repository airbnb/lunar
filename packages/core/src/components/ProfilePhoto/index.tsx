import React from 'react';
import PropTypes from 'prop-types';
import { and, mutuallyExclusiveProps } from 'airbnb-prop-types';
import withStyles, { WithStylesProps } from '../../composers/withStyles';

const mutuallyExclusiveSizePropType = mutuallyExclusiveProps(
  PropTypes.any,
  'small',
  'large',
  'macro',
  'size',
);
const namedSizePropType = and([PropTypes.bool, mutuallyExclusiveSizePropType]);
const unitSizePropType = and([PropTypes.number, mutuallyExclusiveSizePropType]);

export type Props = {
  /** URL of the image to display. */
  imageSrc: string;
  /** Whether the component should be inline. */
  inline?: boolean;
  /** Large size (96px). */
  large?: boolean;
  /** Macro size (160px). */
  macro?: boolean;
  /** Size in units to multiply by. */
  size?: number;
  /** Small size (24px). */
  small?: boolean;
  /** Whether the image should be square (it is rounded by default). */
  square?: boolean;
  /** Accessibility text for the Photo. */
  title: string;
};

/** Display a profile photo. */
export class ProfilePhoto extends React.Component<Props & WithStylesProps> {
  static propTypes = {
    large: namedSizePropType,
    macro: namedSizePropType,
    size: unitSizePropType,
    small: namedSizePropType,
  };

  // Do not supply defaults for the mutually exclusive props,
  // doing so will cause validation errors.
  static defaultProps = {
    inline: false,
    square: false,
  };

  render() {
    const {
      cx,
      imageSrc,
      inline,
      macro,
      large,
      small,
      size,
      square,
      styles,
      title,
      theme,
    } = this.props;
    const { unit } = theme!;

    return (
      <div
        className={cx(
          inline && styles.inline,
          styles.regular,
          macro && styles.macro,
          large && styles.large,
          small && styles.small,
          !!size &&
            size > 0 && {
              height: size * unit,
              maxHeight: size * unit,
              maxWidth: size * unit,
              width: size * unit,
            },
        )}
      >
        <img
          className={cx(
            styles.image,
            styles.regular,
            !square && styles.roundedImage,
            macro && styles.macro,
            large && styles.large,
            small && styles.small,
            !!size &&
              size > 0 && {
                height: size * unit,
                maxHeight: size * unit,
                maxWidth: size * unit,
                width: size * unit,
              },
          )}
          src={imageSrc}
          alt={title}
          title={title}
        />
      </div>
    );
  }
}

export default withStyles(
  ({ color }) => ({
    inline: {
      display: 'inline-block',
    },
    image: {
      display: 'block',
      background: color.core.neutral[6],
      color: color.base,
      objectFit: 'cover',
      overflow: 'hidden',
    },
    roundedImage: {
      borderRadius: '50%',
    },
    small: {
      height: 24,
      width: 24,
      maxHeight: 24,
      maxWidth: 24,
    },
    regular: {
      height: 48,
      width: 48,
      maxHeight: 48,
      maxWidth: 48,
    },
    large: {
      height: 96,
      width: 96,
      maxHeight: 96,
      maxWidth: 96,
    },
    macro: {
      height: 160,
      width: 160,
      maxHeight: 160,
      maxWidth: 160,
    },
  }),
  {
    passThemeProp: true,
  },
)(ProfilePhoto);
