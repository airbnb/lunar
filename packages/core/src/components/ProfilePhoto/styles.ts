import { StyleSheet } from '../../hooks/useStyles';

const styleSheet: StyleSheet = ({ color }) => ({
  inline: {
    display: 'inline-block',
  },

  image: {
    display: 'block',
    background: color.core.neutral[6],
    color: color.base,
    objectFit: 'cover',
    overflow: 'hidden',
  },

  roundedImage: {
    borderRadius: '50%',
  },

  small: {
    height: 24,
    width: 24,
    maxHeight: 24,
    maxWidth: 24,
  },
  regular: {
    height: 48,
    width: 48,
    maxHeight: 48,
    maxWidth: 48,
  },

  large: {
    height: 96,
    width: 96,
    maxHeight: 96,
    maxWidth: 96,
  },

  macro: {
    height: 160,
    width: 160,
    maxHeight: 160,
    maxWidth: 160,
  },
});

export default styleSheet;

export { styleSheet };
