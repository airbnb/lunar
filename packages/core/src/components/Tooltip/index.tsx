/* eslint-disable jsx-a11y/no-static-element-interactions */

import React from 'react';
import uuid from 'uuid/v4';
import Overlay from '../Overlay';
import NotchedBox, { NOTCH_SIZE, NOTCH_SPACING } from '../NotchedBox';
import Text from '../Text';
import withStyles, { WithStylesProps } from '../../composers/withStyles';

export type Props = {
  /** Width of the tooltip in units. */
  width?: number;
  /** What to show in the tooltip. */
  content: NonNullable<React.ReactNode>;
  /** Inline content to hover. */
  children: NonNullable<React.ReactNode>;
  /** True to disable tooltip but still show children. */
  disabled?: boolean;
  /** True to use a light background with dark text. */
  inverted?: boolean;
  /** True to prevent dismissmal on mouse down. */
  remainOnMouseDown?: boolean;
  /** True to add a dotted bottom border. */
  underlined?: boolean;
  /** Callback fired when the tooltip is shown. */
  onShow?: () => void;
};

export type State = {
  labelID: string;
  open: boolean;
  tooltipHeight: number;
  targetRect: ClientRect;
};

export type PositionStruct = {
  above: boolean;
  align: 'left' | 'center' | 'right';
};

export type StyleStruct = {
  center: any;
  right: any;
};

/** A tooltip that renders in an portal, so it can escape potentially overflowed containers. */
export class Tooltip extends React.Component<Props & WithStylesProps, State> {
  static defaultProps = {
    disabled: false,
    inverted: false,
    onShow() {},
    remainOnMouseDown: false,
    underlined: false,
    width: 35,
  };

  state = {
    labelID: uuid(),
    open: false,
    tooltipHeight: 0,
    targetRect: document.body.getBoundingClientRect(),
  };

  containerRef = React.createRef<HTMLSpanElement>();

  currentTooltipRef: HTMLDivElement | null = null;

  rafHandle?: number;

  static getDerivedStateFromProps({ disabled }: Props) {
    if (disabled) {
      return {
        open: false,
      };
    }

    return null;
  }

  componentDidUpdate(prevProps: Props) {
    if (prevProps.content !== this.props.content) {
      this.updateTooltipHeight();
    }
  }

  componentWillUnmount() {
    cancelAnimationFrame(this.rafHandle as number);
  }

  updateTooltipHeight() {
    // we need to know the height of the tooltip to position it above target
    /* istanbul ignore next: refs are hard */
    this.rafHandle = requestAnimationFrame(() => {
      const el = this.currentTooltipRef;
      this.setState({ tooltipHeight: el ? el.offsetHeight : 0 });
    });
  }

  bestPosition(rect: State['targetRect']): PositionStruct {
    const output: PositionStruct = { above: false, align: 'left' };
    const { width: widthProp, theme } = this.props;

    /* istanbul ignore next: refs are hard */
    if (rect && widthProp && theme) {
      const { width: targetWidth, left, right, top } = rect;
      const { unit } = theme;
      const rightSpace = window.innerWidth - right;

      if (top > window.innerHeight * 0.6) {
        output.above = true;
      }

      if (Math.min(left, rightSpace) - unit > (widthProp * unit - targetWidth) / 2) {
        output.align = 'center';
      } else if (rightSpace < left) {
        output.align = 'right';
      }
    }

    return output;
  }

  private handleTooltipRef = (ref: HTMLDivElement | null) => {
    this.currentTooltipRef = ref;
    this.updateTooltipHeight();
  };

  private handleEnter = () => {
    const { current } = this.containerRef;

    /* istanbul ignore if: refs are hard */
    if (current) {
      this.setState({ targetRect: current.getBoundingClientRect() });
    }

    if (!this.props.disabled && !this.state.open) {
      this.setState({ open: true });
      this.props.onShow!();
    }
  };

  private handleMouseDown = () => {
    if (!this.props.remainOnMouseDown) {
      this.handleClose();
    }
  };

  private handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const {
      cx,
      styles,
      theme,
      width: widthProp,
      children,
      content,
      disabled,
      underlined,
      inverted,
    } = this.props;
    const { open, targetRect, tooltipHeight, labelID } = this.state;
    const { unit } = theme!;
    const width = widthProp! * unit;
    const { align, above } = this.bestPosition(targetRect);
    const targetWidth = targetRect.width;
    const halfNotch = (NOTCH_SIZE * unit) / Math.SQRT2;
    const notchOffset: StyleStruct = {
      center: '50%',
      right: -(unit * NOTCH_SPACING + halfNotch),
    };
    const marginLeft: StyleStruct = {
      center: -width / 2 + targetWidth / 2,
      right: -width + targetWidth,
    };
    const distance = halfNotch + 1;

    return (
      <span className={cx(styles.container)} ref={this.containerRef}>
        <div
          aria-labelledby={labelID}
          onMouseEnter={this.handleEnter}
          onMouseLeave={this.handleClose}
          onMouseDown={this.handleMouseDown}
          className={cx(!disabled && underlined && styles.underlined)}
        >
          {children}
        </div>

        <div id={labelID} className={cx(styles.offscreen)}>
          {content}
        </div>

        <Overlay open={open} onClose={this.handleClose} noBackground>
          <div
            role="tooltip"
            className={cx(styles.tooltip, above ? styles.tooltip_above : styles.tooltip_below, {
              width,
              marginLeft: marginLeft[align as keyof StyleStruct],
              marginTop: above ? -(tooltipHeight + targetRect.height + distance) : distance,
              textAlign: align,
            })}
            ref={this.handleTooltipRef}
          >
            <div className={cx(styles.shadow)}>
              <NotchedBox
                inverted={!inverted}
                notchOffset={notchOffset[align as keyof StyleStruct]}
                notchBelow={above}
              >
                <Text inverted={!inverted}>{content}</Text>
              </NotchedBox>
            </div>
          </div>
        </Overlay>
      </span>
    );
  }
}

export default withStyles(
  ({ unit, color, pattern, ui }) => ({
    container: {
      display: 'inline-block',
    },

    offscreen: {
      ...pattern.offscreen,
    },

    underlined: {
      borderBottom: `1px dotted ${color.core.primary[3]}`,
      cursor: 'help',
    },

    tooltip: {
      animationDuration: '200ms',
      animationTimingFunction: 'ease-out',
    },

    tooltip_above: {
      animationName: {
        name: 'fadeDown',
        from: {
          opacity: 0,
          transform: `translateY(${unit * 1.5}px)`,
        },
        to: {
          opacity: 1,
        },
      },
    },

    tooltip_below: {
      animationName: {
        name: 'fadeUp',
        from: {
          opacity: 0,
          transform: `translateY(-${unit * 1.5}px)`,
        },
        to: {
          opacity: 1,
        },
      },
    },

    shadow: {
      display: 'inline-block',
      boxShadow: ui.boxShadowLarge,
      borderRadius: ui.borderRadius,
    },
  }),
  {
    passThemeProp: true,
  },
)(Tooltip);
