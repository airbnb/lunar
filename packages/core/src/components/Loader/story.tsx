import React from 'react';
import Text from '../Text';
import Loader from '.';

export default {
  title: 'Core/Loader',
  parameters: {
    inspectComponents: [Loader],
  },
};

export function aLoaderAbsolutelyCenteredWithinAParent() {
  return (
    <div style={{ height: 100, position: 'relative' }}>
      <Loader />
    </div>
  );
}

aLoaderAbsolutelyCenteredWithinAParent.story = {
  name: 'A loader absolutely centered within a parent.',
};

export function aLoaderInline() {
  return (
    <Text>
      Content before the loader. <Loader inline /> Content after the loader.
    </Text>
  );
}

aLoaderInline.story = {
  name: 'A loader inline.',
};

export function withALargerSizePositionedStatically() {
  return <Loader static large />;
}

withALargerSizePositionedStatically.story = {
  name: 'With a larger size positioned statically.',
};
