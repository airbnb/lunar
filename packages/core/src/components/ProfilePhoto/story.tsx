import React from 'react';
import lunar from ':storybook/images/lunar-logo.png';
import ProfilePhoto from '.';

export default {
  title: 'Core/ProfilePhoto',
  parameters: {
    inspectComponents: [ProfilePhoto],
  },
};

export function defaultProfilePhoto() {
  return <ProfilePhoto imageSrc={lunar} title="Photo" />;
}

defaultProfilePhoto.story = {
  name: 'Default profile photo.',
};

export function withDifferentSizingUnitsBasedSizeSmallRegularDefaultLargeMacro() {
  return (
    <>
      <ProfilePhoto imageSrc={lunar} title="Photo" size={2} />
      <ProfilePhoto small imageSrc={lunar} title="Photo" />
      <ProfilePhoto imageSrc={lunar} title="Photo" />
      <ProfilePhoto large imageSrc={lunar} title="Photo" />
      <ProfilePhoto macro imageSrc={lunar} title="Photo" />
    </>
  );
}

withDifferentSizingUnitsBasedSizeSmallRegularDefaultLargeMacro.story = {
  name: 'With different sizing: units-based size, small, regular (default), large, macro.',
};

export function inlineProfilePhotos() {
  return (
    <>
      <ProfilePhoto inline imageSrc={lunar} title="Photo" size={2} />
      <ProfilePhoto inline small imageSrc={lunar} title="Photo" />
      <ProfilePhoto inline imageSrc={lunar} title="Photo" />
      <ProfilePhoto inline large imageSrc={lunar} title="Photo" />
      <ProfilePhoto inline macro imageSrc={lunar} title="Photo" />
    </>
  );
}

inlineProfilePhotos.story = {
  name: 'Inline profile photos.',
};

export function squareProfilePhoto() {
  return <ProfilePhoto inline large square imageSrc={lunar} title="Photo" />;
}

squareProfilePhoto.story = {
  name: 'Square profile photo.',
};

export function brokenImageWithOverflowingAltText() {
  return <ProfilePhoto large imageSrc="BROKEN_IMAGE" title="Alt title text." />;
}

brokenImageWithOverflowingAltText.story = {
  name: 'Broken image with overflowing alt text.',
};

export function fallbackImageSrcWhenImageIsBroken() {
  return (
    <ProfilePhoto large fallbackImageSrc={lunar} imageSrc="BROKEN_IMAGE" title="Alt title text." />
  );
}

fallbackImageSrcWhenImageIsBroken.story = {
  name: 'Fallback image src when image is broken.',
};
