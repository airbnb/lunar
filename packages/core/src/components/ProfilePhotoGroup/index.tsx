import React from 'react';
import withStyles, { WithStylesProps } from '../../composers/withStyles';
import { Props as ProfilePhotoProps } from '../ProfilePhoto';

/* eslint-disable react/no-array-index-key */

export type Props = {
  /** List of `ProfilePhoto` components to group. */
  children: NonNullable<React.ReactNode>;
  /** Max number of photos to display before truncating. */
  max?: number;
  /** Size in units to multiply by. */
  size?: number;
};

/** Horizontally align `ProfilePhoto`s in a stacked fashion. */
export class ProfilePhotoGroup extends React.Component<Props & WithStylesProps> {
  static defaultProps = {
    max: 3,
    size: 5,
  };

  render() {
    const { cx, children, max, size, styles, theme } = this.props as Required<
      Props & WithStylesProps
    >;
    const { unit } = theme;
    const margin = { marginLeft: -(size * (unit / 2.5)) };
    let photos = React.Children.toArray(children).filter(child => !!child);
    let remainder = 0;

    if (max && photos.length > max) {
      remainder = photos.length - max;
      photos = photos.slice(0, max);
    }

    return (
      <div className={cx(styles.group)}>
        {photos.map((photo, i) => (
          <div key={i} className={cx(styles.cell, margin)}>
            {React.cloneElement(photo as React.ReactElement<ProfilePhotoProps>, {
              size,
              inline: false,
              square: false,
            })}
          </div>
        ))}

        {remainder > 0 && (
          <div key="remainder" className={cx(styles.cell, margin)}>
            <div
              className={cx(styles.remainder, {
                height: size * unit,
                maxHeight: size * unit,
                maxWidth: size * unit,
                width: size * unit,
              })}
            >
              <span className={cx(styles.remainderNumber)}>{`+${remainder}`}</span>
            </div>
          </div>
        )}
      </div>
    );
  }
}

export default withStyles(
  ({ color, font }) => ({
    group: {
      display: 'flex',
      alignItems: 'stretch',
    },

    cell: {
      borderRadius: '50%',
      border: `1px solid ${color.base}`,
      background: color.core.neutral[6],

      ':first-of-type': {
        marginLeft: 0,
      },

      ':empty': {
        display: 'none',
      },
    },

    remainder: {
      ...font.textRegular,
      position: 'relative',
    },

    remainderNumber: {
      transform: 'translate(-50%, -50%)',
      position: 'absolute',
      top: '50%',
      left: '50%',
      color: color.base,
    },
  }),
  {
    passThemeProp: true,
  },
)(ProfilePhotoGroup);
