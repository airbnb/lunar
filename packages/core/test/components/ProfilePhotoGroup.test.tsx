import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import ProfilePhotoGroup from '../../src/components/ProfilePhotoGroup';
import ProfilePhoto from '../../src/components/ProfilePhoto';
import Loader from '../../src/components/Loader';
import Tooltip from '../../src/components/Tooltip';

describe('<ProfilePhotoGroup />', () => {
  const props = {
    imageSrc: 'https://domain.com/some/file.jpg',
    title: 'Name',
  };

  it('errors when a non-button is passed', () => {
    expect(() => {
      shallowWithStyles(
        <ProfilePhotoGroup>
          <Loader />
        </ProfilePhotoGroup>,
      );
    }).toThrowErrorMatchingSnapshot();
  });

  it('doesnt error when `Tooltip` is wrapping', () => {
    expect(() => {
      shallowWithStyles(
        <ProfilePhotoGroup>
          <Tooltip content="">
            <ProfilePhoto {...props} />
          </Tooltip>
        </ProfilePhotoGroup>,
      );
    }).not.toThrowError();
  });

  it('renders photos', () => {
    const wrapper = shallowWithStyles(
      <ProfilePhotoGroup>
        <ProfilePhoto {...props} />
        <ProfilePhoto {...props} />
        <ProfilePhoto {...props} />
      </ProfilePhotoGroup>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a single photo', () => {
    const wrapper = shallowWithStyles(
      <ProfilePhotoGroup>
        <ProfilePhoto {...props} />
      </ProfilePhotoGroup>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a remainder cell when the max is met', () => {
    const wrapper = shallowWithStyles(
      <ProfilePhotoGroup max={1}>
        <ProfilePhoto {...props} />
        <ProfilePhoto {...props} />
        <ProfilePhoto {...props} />
      </ProfilePhotoGroup>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('supports custom sizes', () => {
    const wrapper = shallowWithStyles(
      <ProfilePhotoGroup size={5}>
        <ProfilePhoto {...props} />
        <ProfilePhoto {...props} />
        <ProfilePhoto {...props} />
      </ProfilePhotoGroup>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('handles photos that return falsy values', () => {
    const wrapper = shallowWithStyles(
      <ProfilePhotoGroup>
        {false && <ProfilePhoto {...props} />}
        {true && <ProfilePhoto {...props} />}
        {null && <ProfilePhoto {...props} />}
      </ProfilePhotoGroup>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('handles components that return a falsy value', () => {
    function FakePhoto() {
      return null;
    }

    const wrapper = shallowWithStyles(
      <ProfilePhotoGroup>
        <FakePhoto />
      </ProfilePhotoGroup>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
