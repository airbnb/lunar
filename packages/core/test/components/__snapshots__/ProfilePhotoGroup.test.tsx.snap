// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePhotoGroup /> errors when a non-button is passed 1`] = `
"Warning: Failed prop type: \`ProfilePhotoGroup.children\` only accepts components matching the regular expression /(?:Tooltip|Permission|Trebuchet|ERF)|(?:ProfilePhoto)$/
    in ProfilePhotoGroup"
`;

exports[`<ProfilePhotoGroup /> handles components that return a falsy value 1`] = `
<div
  className="group_1yr212"
>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="0"
  >
    <FakePhoto
      inline={false}
      key=".0"
      size={5}
      square={false}
    />
  </div>
</div>
`;

exports[`<ProfilePhotoGroup /> handles photos that return falsy values 1`] = `
<div
  className="group_1yr212"
>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="0"
  >
    <ProfilePhoto
      imageSrc="https://domain.com/some/file.jpg"
      inline={false}
      key=".1"
      size={5}
      square={false}
      title="Name"
    />
  </div>
</div>
`;

exports[`<ProfilePhotoGroup /> renders a remainder cell when the max is met 1`] = `
<div
  className="group_1yr212"
>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="0"
  >
    <ProfilePhoto
      imageSrc="https://domain.com/some/file.jpg"
      inline={false}
      key=".0"
      size={5}
      square={false}
      title="Name"
    />
  </div>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="remainder"
  >
    <div
      className="remainder_15l21mp-o_O-inline-0_ksaxgv"
    >
      <span
        className="remainderNumber_d1qlh4"
      >
        +2
      </span>
    </div>
  </div>
</div>
`;

exports[`<ProfilePhotoGroup /> renders a single photo 1`] = `
<div
  className="group_1yr212"
>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="0"
  >
    <ProfilePhoto
      imageSrc="https://domain.com/some/file.jpg"
      inline={false}
      key=".0"
      size={5}
      square={false}
      title="Name"
    />
  </div>
</div>
`;

exports[`<ProfilePhotoGroup /> renders photos 1`] = `
<div
  className="group_1yr212"
>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="0"
  >
    <ProfilePhoto
      imageSrc="https://domain.com/some/file.jpg"
      inline={false}
      key=".0"
      size={5}
      square={false}
      title="Name"
    />
  </div>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="1"
  >
    <ProfilePhoto
      imageSrc="https://domain.com/some/file.jpg"
      inline={false}
      key=".1"
      size={5}
      square={false}
      title="Name"
    />
  </div>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="2"
  >
    <ProfilePhoto
      imageSrc="https://domain.com/some/file.jpg"
      inline={false}
      key=".2"
      size={5}
      square={false}
      title="Name"
    />
  </div>
</div>
`;

exports[`<ProfilePhotoGroup /> supports custom sizes 1`] = `
<div
  className="group_1yr212"
>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="0"
  >
    <ProfilePhoto
      imageSrc="https://domain.com/some/file.jpg"
      inline={false}
      key=".0"
      size={5}
      square={false}
      title="Name"
    />
  </div>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="1"
  >
    <ProfilePhoto
      imageSrc="https://domain.com/some/file.jpg"
      inline={false}
      key=".1"
      size={5}
      square={false}
      title="Name"
    />
  </div>
  <div
    className="cell_z0wimt-o_O-inline-0_1j63ti"
    key="2"
  >
    <ProfilePhoto
      imageSrc="https://domain.com/some/file.jpg"
      inline={false}
      key=".2"
      size={5}
      square={false}
      title="Name"
    />
  </div>
</div>
`;
